### PR TITLE
feat: make log to console as optional

### PR DESCRIPTION
**Pull request by jasonyic**
## Issue
harmony only support log to file and `lumberjack` rotation make `ln -sf /dev/stdout zerolog-harmony.log` impossible when running in `docker` and `kubernetes`, which log everything to console.

## Proposed changes
add a boolean configuration `Console` in `Log` to provide the option to log to console only

feature request: #4212

### CODE REVIEW

**Comment by jasonyic** [comment]
#4212

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 TOP:=$(realpath ..)
 export CGO_CFLAGS:=-I$(TOP)/bls/include -I$(TOP)/mcl/include -I/usr/local/opt/openssl/include
 export CGO_LDFLAGS:=-L$(TOP)/bls/lib -L/usr/local/opt/openssl/lib
-export LD_LIBRARY_PATH:=$(TOP)/bls/lib:$(TOP)/mcl/lib:/usr/local/opt/openssl/lib:/opt/homebrew/opt/gmp/lib/:/opt/homebrew/opt/openssl/lib
+export LD_LIBRARY_PATH:=$(TOP)/bls/lib:$(TOP)/mcl/lib:/usr/local/opt/openssl/lib:/opt/homebrew/opt/gmp/lib/:/opt/homebrew/opt/openssl/lib:/usr/lib
```
Comment (MaxMustermann2) [inline Makefile:4]: Why has this been changed?

File: cmd/harmony/config_test.go [hunk 2]
```diff
@@ -59,11 +59,12 @@ Version = "1.0.4"
   Port = 9500
 
 [Log]
+	Console = false
```
Comment (MaxMustermann2) [inline cmd/harmony/config_test.go:62]: nitpick: Please use consistent indentation.

File: cmd/harmony/config_test.go [hunk 3]
```diff
@@ -59,11 +59,12 @@ Version = "1.0.4"
...
   FileName = "harmony.log"
   Folder = "./latest"
   RotateSize = 100
   RotateCount = 0
-  RotateMaxAge = 0
+  RotateMaxAge = 0
```
Comment (MaxMustermann2) [inline cmd/harmony/config_test.go:67]: nit: Please remove the extra spaces.

File: go.sum [hunk 4]
```diff
@@ -508,7 +508,6 @@ github.com/jrick/logrotate v1.0.0/go.mod h1:LNinyqDIJnpAur+b8yyulnQw/wDuN1+BYKlT
 github.com/json-iterator/go v1.1.6/go.mod h1:+SdeFBvtyEkXs7REEP0seUULqWtbJapLOCVDaaPEHmU=
 github.com/json-iterator/go v1.1.7/go.mod h1:KdQUCv79m/52Kvf8AW2vK1V8akMuk1QjK/uOdHXbAo4=
 github.com/json-iterator/go v1.1.8/go.mod h1:KdQUCv79m/52Kvf8AW2vK1V8akMuk1QjK/uOdHXbAo4=
-github.com/json-iterator/go v1.1.10 h1:Kz6Cvnvv2wGdaG/V8yMvfkmNiXq9Ya2KUv4rouJJr68=
```
Comment (MaxMustermann2) [inline go.sum:511]: Why has this been removed?

File: go.sum [hunk 5]
```diff
@@ -870,7 +869,6 @@ github.com/modern-go/concurrent v0.0.0-20180228061459-e0a39a4cb421/go.mod h1:6dJ
 github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd h1:TRLaZ9cD/w8PVh93nsPXa1VrQ6jlwL5oN8l14QlcNfg=
 github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd/go.mod h1:6dJC0mAP4ikYIbvyc7fijjWJddQyLn8Ig3JB5CqoB9Q=
 github.com/modern-go/reflect2 v0.0.0-20180701023420-4b7aa43c6742/go.mod h1:bx2lNnkwVCuqBIxFjflWJWanXIb3RllmbCylyMrvgv0=
-github.com/modern-go/reflect2 v1.0.1 h1:9f412s+6RmYXLWZSEzVVgPGK7C2PphHj5RJrvfx9AWI=
```
Comment (MaxMustermann2) [inline go.sum:873]: Why has this been removed?

File: p2p/stream/protocols/sync/message/msg.pb.go [hunk 6]
```diff
@@ -1,12 +1,13 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.26.0
-// 	protoc        v3.15.8
+// 	protoc-gen-go v1.25.0
```
Comment (MaxMustermann2) [inline p2p/stream/protocols/sync/message/msg.pb.go:3]: Why has this been changed? This PR should not impact this file AFAIK.

**Review by MaxMustermann2 — CHANGES_REQUESTED** [review]
Thanks, just a couple of comments.

**Comment by MaxMustermann2** [comment]
Thank you. Just one last change now, could you add a config file migration like this one?
https://github.com/harmony-one/harmony/blob/main/cmd/harmony/config_migrations.go#L224-L231

**Comment by jasonyic** [comment]
> Thank you. Just one last change now, could you add a config file migration like this one? https://github.com/harmony-one/harmony/blob/main/cmd/harmony/config_migrations.go#L224-L231

done, thank you for taking time to review.

**Comment by jasonyic** [comment]
is this a typo for the rate limit? `RateLimterEnabled`
[RPCOpt]
  DebugEnabled = false
  EthRPCsEnabled = true
  LegacyRPCsEnabled = true
  RateLimterEnabled = true

**Review by MaxMustermann2 — APPROVED** [review]

**Comment by MaxMustermann2** [comment]
Thanks for your PR!

> is this a typo for the rate limit?

Appears to be, but it seems to be consistently used throughout the config file and the source code so I wouldn't bother too much with it.

**Review by Frozen — APPROVED** [review]

**Comment by jasonyic** [comment]
> Thanks for your PR!
> 
> > is this a typo for the rate limit?
> 
> Appears to be, but it seems to be consistently used throughout the config file and the source code so I wouldn't bother too much with it.

got it.

**Review by rlan35 — APPROVED** [review]

**Pull request merged**